queue_processors/autoreduction_processor/autoreduction_processor.py

```diff
@@ -137,33 +136,22 @@ def __init__(self):
         """ Initialise consumer. """
         self.consumer_name = "queueProcessor"
 
-    def run(self):
+    @staticmethod
+    def run():
         """
-        Connect to ActiveMQ and listen to the queue for messages.
+        Connect to ActiveMQ via the QueueClient and listen to the
+        /ReductionPending queue for messages.
         """
-        brokers = [(ACTIVEMQ['brokers'].split(':')[0], int(ACTIVEMQ['brokers'].split(':')[1]))]
-        connection = stomp.Connection(host_and_ports=brokers, use_ssl=False)
-        connection.set_listener(self.consumer_name, Listener(connection))
-        logger.info("Starting ActiveMQ Connection to %s", ACTIVEMQ['brokers'])
-        logger.info("Completed ActiveMQ Connection")
-        connection.connect(ACTIVEMQ['amq_user'],
-                           ACTIVEMQ['amq_pwd'],
-                           wait=False,
-                           header={'activemq.prefetchSize': '1'})
-        for queue in ACTIVEMQ['amq_queues']:
-            connection.subscribe(destination=queue,
-                                 id='1',
-                                 ack='client-individual',
-                                 header={'activemq.prefetchSize': '1'})
-            logger.info("[%s] Subscribing to %s", self.consumer_name, queue)
-        logger.info("Successfully subscribed to all of the queues")
-
-        # activemq_client = QueueClient()
-        # activemq_client.connect()
+        activemq_client = QueueClient()
+        connection = activemq_client.connect()
 
         # Create event listener
```

Comment:
Interesting that this code was already here but commented out...

Reply:
Yes, they were there from my first attempt a long time ago to get this right